### DISCUSSION TITLE
Not sure if this is related to Explorer Patcher

### DISCUSSION THREAD

**Discussion by Anixx** [valinet/ExplorerPatcher#415] [General]
There is a strange line at the top of the PowerShell window:
![изображение](https://user-images.githubusercontent.com/1271736/142502423-2a821dff-1412-43df-b546-b928f02e6f7b.png)

**Comment by Anixx** [comment]
The same line appears in other console apps, such as Winevdm installer.

**Reply by valinet** [reply]
Yeah, latest version? It’s not “strange”… I thought I removed all debug info, maybe I forgot this one. So, latest version or not?

**Reply by valinet** [reply]
You know, grep on the source code is always an option. I think I removed that a couple of versions ago, why report old stuff…?

https://github.com/valinet/ExplorerPatcher/commit/1608abf6c6fe3c77b973107cad989aeeb9b1a179#diff-a1aebd672b1921bfb4fcd82196a381f0471ddd320e030ebe1ec984735aded8fa

**Reply by Anixx** [reply]
This is with version 38.2, which is currently the latest.

**Reply by valinet** [reply]
No, it's not, the commit shows it clearly that it was removed. Figure it out, please, it's not that difficult to check the file versions a bit...

![image](https://user-images.githubusercontent.com/6503598/142591241-5b592c41-ded3-44b5-9ae1-0f7d763f38e1.png)

**Reply by Anixx** [reply]
the dxgi.dll is 38.2, the shell extensins are 36.1. The updater did not update the shell extension.

**Reply by valinet** [reply]
So, it was like I said. 36.2 is ages ago. The updater has been introduced in 37.0, along with option to register the shell extension from EP. if you use that, it will get updated, if you use the shell extension from some custom location as before, how can the updater know about it...? I mean, it can, but it just doesn't care, it updates DLLs in `%PROGRAMFILES%\ExplorerPatcher` no matter what. So tl;dr register the shell extension from EP and reboot the computer eventually, so the new thing gets loaded.

You can verify the path to the DLL that is registered to be used in `HKLM\SOFTWARE\Classes\CLSID\{D17F1E1A-5919-4427-8F89-A1A8503CA3EB}\InProcServer32`.

**Reply by Anixx** [reply]
All the files are in c:\windows. 
![изображение](https://user-images.githubusercontent.com/1271736/142597560-4f7056ef-1aa3-4acb-914f-62c4b5db08db.png)
Still, it is not touched by the updater.

**Reply by valinet** [reply]
THE UPDATE PLACES THE FILES IN `%PROGRAMFILES%\ExplorerPatcher`. IT'S HARDCODED FOR THAT. USE THE OPTION IN THE APP TO REGISTER AS SHELL EXTENSION. THAT WILL UPDATE THAT PATH IN THE REGISTRY TO `%PROGRAMFILES%\ExplorerPatcher\ExplorerPatcher.amd64.dll`. THAT GETS UPDATED BY THE UPDATER !!!

**Reply by valinet** [reply]
Check the source code as well!

**Reply by Anixx** [reply]
I re-registered the files, and now there is no that line in powershell!
I did not even know that it moved to Program Files...

**Reply by Anixx** [reply]
>  USE THE OPTION IN THE APP TO REGISTER AS SHELL EXTENSION. THAT WILL UPDATE THAT PATH IN THE REGISTRY TO %PROGRAMFILES%\ExplorerPatcher

No, it did not. I enabled that option as soon as it became available, but the path remained pointing to C:\Windows.

**Reply by valinet** [reply]
Yeah, other than the release notes, the source code itself or more or less randomly noticing, how else can you know about it? I don't do any skywriting or other form of public announcements. That's why I always say check the source and I always get ignored.

**Reply by valinet** [reply]
> 
> 
> > USE THE OPTION IN THE APP TO REGISTER AS SHELL EXTENSION. THAT WILL UPDATE THAT PATH IN THE REGISTRY TO %PROGRAMFILES%\ExplorerPatcher
> 
> No, it did not. I enabled that option as soon as it became available, but the path remained pointing to C:\Windows.

Whatever, I tested that, it seems to work. Maybe it glitched out, maybe something else happened. At least it works now. The path in "Program Files" won't change. Thx.
